src/Test/QuickCheck/Arbitrary.hs

```diff
@@ -723,7 +723,7 @@ instance Arbitrary Float where
     where
       smallDenominators = sized $ \n -> do
         i <- chooseInt (0, n)
-        pure (fromRational (streamNth i rationalUniverse))
+        pure (fromRational (streamNth (min i 256) rationalUniverse))
```

Comment:
This creates a strange distribution of `min i 256`. Better to do `i <- chooseInt (0, min n 256)`.

Reply:
Yeah, I wasn't sure about that. I went with this on grounds "larger size means you still get bigger things here on average", but happy to change.

Reply:
Done.

Reply:
The problem with the old code is that larger size => n is almost always bigger than 256 => the result is almost always the same.

Reply:
Oh yeah, that makes sense.